numpyro/examples/baseball.py

```diff
@@ -0,0 +1,209 @@
+    model_trace = trace(model).get_trace(at_bats)
+    predictions = model_trace['obs']['value']
+    print_results('=' * 30 + header + '=' * 30,
+                  predictions,
+                  player_names,
+                  at_bats,
+                  hits)
+    if not train:
+        model = substitute(model, z)
+        model_trace = trace(model).get_trace(at_bats, hits)
+        log_joint = 0.
+        for site in model_trace.values():
+            site_log_prob = site['fn'].log_prob(site['value'])
+            log_joint = log_joint + onp.apply_over_axes(np.sum,
+                                                        site_log_prob,
+                                                        range(1, np.ndim(site_log_prob)))
+        log_pred_density = logsumexp(log_joint) - np.log(np.shape(log_joint)[0])
+        print('\nLog predictive density: {:.2f}\n'.format(log_pred_density))
+
+
+def predictive_log_density(model, at_bats, z, rng):
+    model = substitute(seed(model, rng), z)
+    model_trace = trace(model).get_trace(at_bats)
+    log_joint = 0.
+    for site in model_trace.values():
+        if site['type'] == 'sample':
+            log_joint = log_joint + np.sum(site['fn'].log_prob(site['value']))
+    return log_joint, model_trace
+
+
+def print_results(header, preds, player_names, at_bats, hits):
+    columns = ['', 'At-bats', 'ActualHits', 'Pred(p25)', 'Pred(p50)', 'Pred(p75)']
+    header_format = '{:>20} {:>10} {:>10} {:>10} {:>10} {:>10}'
+    row_format = '{:>20} {:>10.0f} {:>10.0f} {:>10.2f} {:>10.2f} {:>10.2f}'
+    quantiles = onp.quantile(preds, [0.25, 0.5, 0.75], axis=0)
+    print('\n', header, '\n')
+    print(header_format.format(*columns))
+    for i, p in enumerate(player_names):
+        print(row_format.format(p, at_bats[i], hits[i], *quantiles[:, i]), '\n')
+
+
+def main(args):
+    _, fetch_train = load_dataset(BASEBALL, split='train', shuffle=False)
+    train, player_names = fetch_train()
+    _, fetch_test = load_dataset(BASEBALL, split='test', shuffle=False)
+    test, _ = fetch_test()
+    at_bats, hits = train[:, 0], train[:, 1]
+    season_at_bats, season_hits = test[:, 0], test[:, 1]
+    for i, model in enumerate((fully_pooled,
+                               not_pooled,
+                               partially_pooled,
+                               partially_pooled_with_logit,
+                               )):
+        rng, rng_predict = random.split(random.PRNGKey(i))
+        zs = run_inference(model, at_bats, hits, random.PRNGKey(i), args)
+        predict(model, at_bats, hits, zs, rng_predict, player_names)
+        predict(model, season_at_bats, season_hits, zs, rng_predict, player_names, train=False)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Baseball batting average using HMC")
+    parser.add_argument("-n", "--num-samples", nargs="?", default=3000, type=int)
+    parser.add_argument("--num-warmup-steps", nargs='?', default=1500, type=int)
```

Comment:
Not for this PR: should we change this arg in hmc to `warmup_steps`?

Reply:
Sure, I think there was no other reason to name it `num_warmup_steps` than to have a similar name to `num_samples`. Maybe we can also rename it as `num_warmup`. :)